layersvt/api_dump.h

```diff
@@ -241,9 +243,96 @@ class ConditionalFrameOutput {
     }
 };
 
+#ifdef __ANDROID__
+template <class char_type = char, class traits = std::char_traits<char_type>>
+class AndroidLogcatBuf final : public std::basic_streambuf<char_type, traits> {
+   public:
+    class LogWriter {
+       public:
+        virtual void write(const std::basic_string<char_type, traits> &data) = 0;
+        virtual ~LogWriter() {}
+    };
+
+    AndroidLogcatBuf(std::unique_ptr<LogWriter> log_writer, size_t bufsize = 0x400)
```

Comment:
I need to change the default `bufsize` to a smaller value like `0x200`, to prevent `__android_log_print` from truncating the message, we can easily hit this issue if we ask the api dump layer to output the shader source.

`__android_log_print` will truncate any input longer than 1024 because [`__android_log_print`](https://cs.android.com/android/platform/superproject/main/+/e60a2cd5ccf434f66761ab1bcd22169ab6372be2:system/logging/liblog/logger_write.cpp;l=410) [uses  `vsnprintf`](https://cs.android.com/android/platform/superproject/main/+/e60a2cd5ccf434f66761ab1bcd22169ab6372be2:system/logging/liblog/logger_write.cpp;l=421) with `LOG_BUF_SIZE` as the length parameter to format the input string. And `LOG_BUF_SIZE` is currently [defined as 1024](https://cs.android.com/android/platform/superproject/main/+/e60a2cd5ccf434f66761ab1bcd22169ab6372be2:system/logging/liblog/logger_write.cpp;l=65).

Reply:
Done.